ludwig/features/image_feature.py

```diff
@@ -90,6 +92,32 @@
 ###
 # Image specific augmentation operations
 ###
+@register_augmentation_op(name="auto_augmentation", features=IMAGE)
+class AutoAugment(torch.nn.Module):
+    def __init__(self, config: AutoAugmentationConfig):
+        super().__init__()
+        self.auto_augmentation_method = config.method
+        self.augmentation_method = self.get_augmentation_method()
+
+    def get_augmentation_method(self):
+        match self.auto_augmentation_method:
```

Comment:
Hey @ethanreidel! Ludwig currently doesn't support Py 3.11 and beyond. I believe the `match` and `case` statements were only introduced in Py311, so this is probably why the tests are failing. Are you able to use conditionals instead?

Reply:
Hi @arnavgarg1. Yeah you're 100% correct. Will swap to conditionals. 👍

Reply:
Also @arnavgarg1, regarding the question you asked about benchmark performance. I have tested all three auto-augmentation methods (auto_augment, trivial_augment, and rand_augment) on the MNIST dataset example. The auto augment method tended to perform the best of the three, however, I have yet to test with other datasets.

Reply:
@ethanreidel Nice! Good to know :) Great work on this and I'm excited for this to land!